search5/src/main/java/org/hibernate/performance/search/model/application/SearchProgrammaticMapping.java

```diff
@@ -0,0 +1,149 @@
+		owner.indexEmbedded().depth( 1 );
+		owner.containedIn();
+
+		// Employee
+		IndexedMapping employee = mapping.entity( Employee.class ).indexed();
+		employee
+				.property( "id", ElementType.FIELD ).documentId()
+				.property( "name", ElementType.FIELD ).field().analyze( Analyze.NO )
+				.property( "surname", ElementType.FIELD ).field().analyze( Analyze.NO )
+				.property( "socialSecurityNumber", ElementType.FIELD ).field().analyze( Analyze.NO )
+				.property( "company", ElementType.FIELD ).indexEmbedded()
+				.property( "businessUnit", ElementType.FIELD ).indexEmbedded();
+
+		PropertyMapping managerProperty = employee.property( "manager", ElementType.FIELD );
+		// index 4 managers up to the hierarchy
+		managerProperty.indexEmbedded().depth( 4 );
+		managerProperty.containedIn();
+
+		// Manager
+		IndexedMapping manager = mapping.entity( Manager.class ).indexed();
+		PropertyMapping employees = manager.property( "employees", ElementType.FIELD );
+		// index 1 employee-levels down to the hierarchy
+		employees.indexEmbedded().depth( 1 );
+		employees.containedIn();
+
+		// QuestionnaireDefinition
+		IndexedMapping questionnaireDefinition = mapping.entity( QuestionnaireDefinition.class ).indexed();
+		questionnaireDefinition.property( "id", ElementType.FIELD ).documentId();
+		questionnaireDefinition.property( "title", ElementType.FIELD ).field();
+		questionnaireDefinition.property( "description", ElementType.FIELD ).field();
+		questionnaireDefinition.property( "year", ElementType.FIELD ).field().numericField().sortableField();
+		questionnaireDefinition.property( "company", ElementType.FIELD ).indexEmbedded();
+
+		PropertyMapping questions = questionnaireDefinition.property( "questions", ElementType.FIELD );
+		questions.indexEmbedded().depth( 1 );
+		questions.containedIn();
+
+		// Question
+		IndexedMapping question = mapping.entity( Question.class ).indexed();
+		question.property( "id", ElementType.FIELD ).documentId();
+		question.property( "text", ElementType.FIELD ).field();
+
+		PropertyMapping questionnaire = question.property( "questionnaire", ElementType.FIELD );
+		questionnaire.indexEmbedded().depth( 1 );
+		questionnaire.containedIn();
+
+		// OpenQuestion
+		mapping.entity( OpenQuestion.class ).indexed();
+
+		// ClosedQuestion
+		IndexedMapping closedQuestion = mapping.entity( ClosedQuestion.class ).indexed();
+		closedQuestion.property( "weight", ElementType.FIELD ).field().numericField();
+
+		// QuestionnaireInstance
+		IndexedMapping questionnaireInstance = mapping.entity( QuestionnaireInstance.class ).indexed();
+		questionnaireInstance
+				.property( "id", ElementType.FIELD ).documentId()
+				.property( "uniqueCode", ElementType.FIELD ).field().analyze( Analyze.NO )
+				.property( "definition", ElementType.FIELD ).indexEmbedded()
+				.property( "approval", ElementType.FIELD ).indexEmbedded()
+				.property( "subject", ElementType.FIELD ).indexEmbedded()
```

Comment:
If you do this, I think there should be a `containedIn()` from `Employee` to `QuestionnaireInstance`. Employees are expected to have their data change from time to time, and it wouldn't cost much to reindex their questionnaires when that happens.